lis/surfacemodels/land/ac.7.2/ac72_lsmMod.F90

```diff
@@ -0,0 +1,248 @@
+!  \end{description}
+!EOP
+        implicit none        
+        integer  :: n, t, i    
+        integer  :: status   
+
+        ! allocate memory for nest 
+        allocate(AC72_struc(LIS_rc%nnest))
+ 
+        ! read configuation information from lis.config file
+        call AC72_readcrd()
+
+        do n=1, LIS_rc%nnest
+            ! allocate memory for all tiles in current nest 
+            allocate(AC72_struc(n)%ac72(LIS_rc%npatch(n, LIS_rc%lsm_index)))
+            !------------------------------------------------------------------------
+            ! allocate memory for vector variables passed to model interfaces        
+            ! TODO: check the following allocation statements carefully!
+            !------------------------------------------------------------------------
+            AC72_struc(n)%max_No_compartments = 12 ! hard coded
+            do t=1, LIS_rc%npatch(n, LIS_rc%lsm_index)
+                allocate(AC72_struc(n)%ac72(t)%smc(AC72_struc(n)%max_No_compartments))
+                allocate(AC72_struc(n)%ac72(t)%tmincli_monthly(12))
+                allocate(AC72_struc(n)%ac72(t)%tmaxcli_monthly(12))
+            enddo
+
+            ! allocate memory for Trecord arrays
+            do t=1,LIS_rc%npatch(n, LIS_rc%lsm_index)
+                allocate(AC72_struc(n)%ac72(t)%Tmax_record(366))
+                allocate(AC72_struc(n)%ac72(t)%Tmin_record(366))
+            enddo
+            
+            ! initialize forcing variables to zeros
+            do t=1, LIS_rc%npatch(n, LIS_rc%lsm_index)
+                AC72_struc(n)%ac72(t)%tair = 0.0
+                AC72_struc(n)%ac72(t)%tmax = 0.0
+                AC72_struc(n)%ac72(t)%tmin = 0.0
+                AC72_struc(n)%ac72(t)%tdew = 0.0
+                AC72_struc(n)%ac72(t)%wndspd = 0.0            
+                AC72_struc(n)%ac72(t)%psurf = 0.0
+                AC72_struc(n)%ac72(t)%prcp = 0.0
+                AC72_struc(n)%ac72(t)%eto = 0.0
+
+                !LB: Initialize HarvestNow (new from AC7.1)
+                AC72_struc(n)%ac72(t)%HarvestNow = .false.
+                
+            enddo ! end of tile (t) loop
+!------------------------------------------------------------------------
+! Model timestep Alarm
+!------------------------------------------------------------------------
+            AC72_struc(n)%forc_count = 0
+
+            call LIS_update_timestep(LIS_rc, n, AC72_struc(n)%ts)
+
+            call LIS_registerAlarm("AC72 model alarm",&
+                                   AC72_struc(n)%ts, &
+                                   AC72_struc(n)%ts)
+
+            call LIS_registerAlarm("AC72 restart alarm", &
+                                   AC72_struc(n)%ts,&
+                                   AC72_struc(n)%rstInterval)
+            ! Set number of soil moisture layers in surface model (corresponding to
+            ! the compartments for AquaCrop)
+            LIS_sfmodel_struc(n)%nsm_layers = AC72_struc(n)%max_No_compartments
```

Comment:
It is used for the writing of the output. And this is where it might get confusing because for AquaCrop, the comp sizes are defined in LDT
![image](https://github.com/user-attachments/assets/6cd67aa0-0e87-47fe-b273-0f0451aec22c)